src/mcedit2/panels/worldmeta.py

```diff
@@ -0,0 +1,255 @@
+"""
+    player
+"""
+from __future__ import absolute_import
+import logging
+from contextlib import contextmanager
+
+from PySide import QtGui
+from PySide.QtCore import Qt
+
+from mcedit2.util.resources import resourcePath
+from mcedit2.util.load_ui import load_ui
+from mcedit2.util.screen import centerWidgetInScreen
+from mcedit2.widgets.nbttree.nbteditor import NBTDataChangeCommand
+
+log = logging.getLogger(__name__)
+
+
+class WorldMetaPanel(QtGui.QWidget):
+    GENERATOR_TYPES = ["default", "flat", "largeBiomes", "amplified", "customized", 'debug_all_block_states']
+    GAMEMODES = ["Survival", "Creative", "Adventure", "Spectator"]
+    DIFFICULTIES = ["Peaceful", "Easy", "Normal", "Hard"]
+
+    editsDisabled = False
+
+    def __init__(self, editorSession):
+        super(WorldMetaPanel, self).__init__(QtGui.qApp.mainWindow, f=Qt.Tool)
+        self.editorSession = editorSession
+        self.worldMeta = editorSession.worldEditor.adapter.metadata
```

Comment:
`adapter.metadata` will change whenever the revision changes, so you need to get it again in `revisionDidChange` 

In fact, we shouldn't need to go through `adapter` for anything. The WorldEditor should be providing everything.